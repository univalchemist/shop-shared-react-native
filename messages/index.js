import { en_HK as shop_en_HK, zh_HK as shop_zh_HK } from '@components/shops';
import en_HK from './en-HK.json';
import en_UK from './en-UK.json';
import id_ID from './id-ID.json';
import zh_HK from './zh-HK.json';

const messages = {
  'en-HK': { ...en_HK, ...shop_en_HK },
  'en-UK': en_UK,
  'id-ID': id_ID,
  'zh-HK': { ...zh_HK, ...shop_zh_HK },
};

export default messages;
export { en_HK, en_UK, zh_HK, id_ID };
