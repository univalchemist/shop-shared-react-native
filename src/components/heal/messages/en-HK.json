{
  "available": "Available",
  "doctor": "Doctor",
  "doctor.bookAppointment": "Book an appointment",
  "doctor.scheduleAppointment": "Schedule appointment",
  "doctor.scheduleAppointment.selectFamilyMember": "Select family member",
  "doctor.scheduleAppointment.selectClinic": "Select Clinic",
  "doctor.scheduleAppointment.pickDates": "Pick Dates",
  "doctor.scheduleAppointment.maxDates": "You can add up to 3 dates",
  "doctor.scheduleAppointment.addDates": "Add Consultations dates",
  "doctor.scheduleAppointment.alert": "Are you sure you want to delete this date ?",
  "doctor.scheduleAppointment.delete": "Delete",
  "doctor.scheduleAppointment.cancel": "Cancel",
  "doctor.scheduleAppointment.pickDate": "Pick your dates",
  "doctor.scheduleAppointment.pickTime": "Pick your time",
  "doctor.scheduleAppointment.confirmDate": "Confirm Date",
  "doctor.scheduleAppointment.time.am": "AM",
  "doctor.scheduleAppointment.time.pm": "PM",
  "doctor.scheduleAppointment.confirmation": "Please confirm the information of the patient before checking-in.",
  "doctor.scheduleAppointment.next": "Next",
  "doctor.scheduleAppointment.pickDate.error": "You already have an appointment for this date Please pick another date",
  "doctor.scheduleAppointment.requestAppointment": "Request appointment",
  "doctor.scheduleAppointment.appointmentRequested": "Appointment requested",
  "doctor.scheduleAppointment.scheduleConfirmation": "You will receive a notification later to confirm the details of the appointment.",
  "doctor.scheduleAppointment.viewAppointment": "View my appointments",
  "doctor.scheduleAppointment.back": "Back to doctor details",
  "doctorSearch.placeholder": "Search doctor or location",
  "doctor.found": "{total} doctors found",
  "doctor.workIn": "Works in {number} clinic(s)",
  "unavailable": "Unavailable",
  "speciality.name.generalpractitio": "General Practitioner",
  "speciality.name.generalsurgery": "General Surgery",
  "speciality.name.gastroenterology": "Gastroenterology & Hepatology",
  "speciality.name.administrativeme": "Administrative Medicine",
  "speciality.name.anaesthesiology": "Anaesthesiology",
  "speciality.name.cardiology": "Cardiology",
  "speciality.name.cardiothoracicsu": "Cardio-thoracic Surgery",
  "speciality.name.chiropractor": "Chiropractor",
  "speciality.name.clinicalmicrobio": "Clinical Microbiology & Infection",
  "speciality.name.clinicaloncology": "Clinical Oncology",
  "speciality.name.communitymedicin": "Community Medicine",
  "speciality.name.criticalcaremedi": "Critical Care Medicine",
  "speciality.name.dentist": "Dentist",
  "speciality.name.dermatologyvener": "Dermatology & Venereology",
  "speciality.name.developmentalbeh": "Developmental-Behavioural Paediatrics",
  "speciality.name.emergencymedicin": "Emergency Medicine",
  "speciality.name.counsellingpsych": "Counseling Psychology",
  "speciality.name.geneticsandgenom": "Genetics and Genomics (Pediatric)",
  "speciality.name.paediatrics": "Paediatrics",
  "speciality.name.urogynaecology": "Urology",
  "speciality.name.orthopaedicstrau": "Orthopaedics & Traumatology",
  "specialist": "Specialist",
  "specialist.placeholder": "Search for a doctor",
  "findSpecialist": "Find a specialist",
  "clinic": "Clinic",
  "clinic.pluralFound": "{number} clinics found",
  "clinic.singularFound": "{number} clinic found",
  "clinic.noClinics": "No clinics available",
  "clinic.noClinicsDescription": "There are no clinics matching your search criteria.",
  "clinic.pluralDoctor": "{number} Doctors",
  "clinic.singularDoctor": "{number} Doctor",
  "heal.clinic.mdistance": "{number}m away",
  "heal.clinic.kmdistance": "{number}km away",
  "clinic.doctors": "Doctors",
  "clinic.clinicStatus": "Clinic Status",
  "clinic.queueButton": "Get a queue",
  "clinic.open": "Open",
  "clinic.closed": "Closed",
  "clinic.estimatedTime": "estimated consulation time",
  "clinic.openingHours.Sun": "Sunday",
  "clinic.openingHours.Mon": "Monday",
  "clinic.openingHours.Tue": "Tuesday",
  "clinic.openingHours.Wed": "Wednesday",
  "clinic.openingHours.Thu": "Thursday",
  "clinic.openingHours.Fri": "Friday",
  "clinic.openingHours.Sat": "Saturday",
  "clinic.openingHours.Public Holiday": "Public Holiday",
  "heal.NearMe": "Near me",
  "heal.UnifySearch.NotFound": "There are no clinics or doctors matching your search criteria.",
  "heal.UnifySearch.placeholder": "Search clinic or location",
  "heal.UnifySearch.DoctorName": "Doctor name",
  "heal.UnifySearch.ClinicName": "Clinic name",
  "heal.UnifySearch.Location": "Location",
  "heal.UnifySearch.Speciality": "Speciality",
  "heal.UnifySearch.Recent": "Recent",
  "heal.checkin.SelectMember": "Select family member",
  "heal.checkin.NoticeConfirmation": "Please confirm the information of the patient before checking-in.",
  "heal.next": "Next",
  "heal.checkin.title": "Check-in",
  "thankYouScreen.headText": "Thank you for booking with us!",
  "thankYouScreen.timeText": "Estimation consultation time is at {time} on {date}",
  "thankYouScreen.buttonText": "View my appointments",
  "appointmentDetailScreen.headText": "The Clinic Group @Fusionopolis",
  "appointmentDetailScreen.rejectBtnText": "Reject",
  "appointmentDetailScreen.acceptBtnText": "Accept",
  "appointmentDetailScreen.upComingBtnText": "Cancel appointment",
  "appointmentConfirmation.pendingConfirmationTittle": "Appointment is Confirmed",
  "appointmentConfirmation.pendingConfirmationMessage": "Your appointment schedule has been confirmed!",
  "appointmentConfirmation.rejectTittle": "Confirmation",
  "appointmentConfirmation.rejectMessage": "Are you sure you want to cancel this appointment schedule?",
  "appointmentConfirmation.confirm": "Confirm",
  "appointmentConfirmation.cancel": "Cancel",
  "appointmentConfirmation.Ok": "Ok",
  "healTabNavigator.findDoctor": "Find a doctor",
  "healTabNavigator.appointment": "Appointments",
  "healTabNavigator.medication": "Medications",
  "heal.CheckInForm.fullname.label": "Fullname (as in NRIC / HK ID)*",
  "heal.CheckInForm.chineseFullname.label": "Chinese fullname",
  "heal.CheckInForm.idNumber.label": "Identification number*",
  "heal.CheckInForm.phoneNumber.label": "Phone number",
  "heal.CheckInForm.email.label": "Email address",
  "heal.CheckInForm.dateOfBirth.label": "Date of birth*",
  "heal.CheckInForm.address.label": "Address",
  "heal.CheckInForm.contactPerson.label": "Contact person",
  "heal.CheckInForm.relationship.label": "Relationship",
  "heal.CheckInForm.contactNumber.label": "Contact number",
  "heal.CheckInForm.description": "A remote ticketing deposit of $1 will be charged. It will be refunded after your doctorâ€™s visit. If you cancel withn 60min of your estimated consultation time or no-show, the deposit will not be refunded.",
  "heal.myself": "Myself",
  "heal.CheckInForm.required": "This field is required",
  "heal.gender.male": "Male",
  "heal.gender.female": "Female",
  "heal.gender": "Gender",
  "heal.walkInCheckIn.back": "Back to homepage",
  "heal.walkInCheckIn.estimatedConsultation": "Estimated consulation time",
  "heal.walkInCheckIn.title": "You're checked-in",
  "heal.walkInSelectDoctor.selectTitle": "Select your doctor",
  "heal.walkInSelectDoctor.confirmButton": "Confirm",
  "appointment.doctorPrefix": "Dr.",
  "appointmentListingScreen.noData": "No appointments yet.",
  "heal.CheckInForm.confirmAndNext": "Confirm & Next",
  "heal.backToHomepage": "Back to homepage",
  "heal.confirmation.checkedIn": "You're checked-in",
  "heal.CheckIn.estConsultationTime": "Estimated consultation time",
  "heal.CheckIn.notice": "Please check the details of your appointment today.",
  "heal.CheckIn.seeAnotherDoctors": "See another doctor",
  "heal.confirmNextScreen.heading": "Please check the detail of your appointment today.",
  "heal.confirmNextScreen.subHeading": "Appointment Schedule",
  "heal.confirmNextScreen.anotherDoctor": "See another doctor",
  "heal.confirmNextScreen.btnText": "Confirm & Next"
}
