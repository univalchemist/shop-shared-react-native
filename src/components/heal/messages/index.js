import en_HK from './en-HK.json';
import zh_HK from './zh-HK.json';

const messages = {
  'en-HK': en_HK,
  'zh-HK': zh_HK,
};

export default messages;
export { en_HK, zh_HK };
