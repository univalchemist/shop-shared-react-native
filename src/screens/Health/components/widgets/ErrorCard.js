import React from 'react';
import { TouchableOpacity } from 'react-native';
import { useNavigation } from '@react-navigation/native';
import { CardContainer } from '@screens/Health/components/widgets/CardContainer';
import { useTheme, useIntl } from '@wrappers/core/hooks';
import {
  Box,
  ErrorText,
  SectionHeadingText,
  Button,
} from '@wrappers/components';
import { LIFESTYLE_FORM } from '@routes';

export const ErrorCard = ({ color }) => {
  const navigation = useNavigation();
  const theme = useTheme();
  const intl = useIntl();

  return (
    <TouchableOpacity
      activeOpacity={1}
      onPress={() =>
        navigation.navigate(LIFESTYLE_FORM, { scrollToFutureMe: true })
      }
    >
      <CardContainer
        theme={theme}
        justifyContent="center"
        color={color}
        flexGrow={1}
      >
        <SectionHeadingText textAlign="center">
          {intl.formatMessage({
            id: 'health.healthScoreHistory.errorMessage.somethingWentWrong',
          })}
        </SectionHeadingText>

        <ErrorText
          mt={2}
          textAlign="center"
          accessible={true}
          accessibilityLabel={intl.formatMessage({
            id: 'health.lifestyleimage.errorMessage.unableToShow',
          })}
          flexGrow={1}
        >
          {intl.formatMessage({
            id: 'health.lifestyleimage.errorMessage.unableToShow',
          })}
        </ErrorText>
        <ErrorText
          textAlign="center"
          accessible={true}
          accessibilityLabel={intl.formatMessage({
            id: 'health.lifestyleimage.errorMessage.tryAgain',
          })}
          flexGrow={1}
        >
          {intl.formatMessage({
            id: 'health.lifestyleimage.errorMessage.tryAgain',
          })}
        </ErrorText>
        <Box mt={4}>
          <Button
            primary
            title={intl.formatMessage({ id: 'tryAgain' })}
            onPress={() =>
              navigation.navigate(LIFESTYLE_FORM, { scrollToFutureMe: true })
            }
          />
        </Box>
      </CardContainer>
    </TouchableOpacity>
  );
};
