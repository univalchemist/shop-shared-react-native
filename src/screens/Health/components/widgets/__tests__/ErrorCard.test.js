import React from 'react';
import { ErrorCard } from '../ErrorCard';
import { renderForTest } from '@testUtils';
import messages from '@messages/en-HK.json';

describe('Error Card', () => {
  let errorCard = renderForTest(<ErrorCard color={'#FFF'} />);

  it.skip('should match the snapshot', () => {
    expect(errorCard.toJSON()).toMatchSnapshot();
  });

  it('should render text Unable to show data', () => {
    const unableToShowError = errorCard.queryAllByText(
      messages['health.lifestyleimage.errorMessage.unableToShow'],
    );

    expect(unableToShowError.length).toBe(1);
  });

  it('should render text Try again later', () => {
    const tryAgainError = errorCard.queryAllByText(
      messages['health.lifestyleimage.errorMessage.tryAgain'],
    );

    expect(tryAgainError.length).toBe(1);
  });
});
