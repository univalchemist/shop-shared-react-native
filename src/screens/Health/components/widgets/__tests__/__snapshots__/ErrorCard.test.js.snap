// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Error Card should match the snapshot 1`] = `
<View
  alignItems="center"
  color="#FFF"
  flexGrow={1}
  justifyContent="center"
  style={
    Array [
      Object {
        "alignItems": "center",
        "backgroundColor": "#ffffff",
        "borderRadius": 4,
        "borderTopColor": "#FFF",
        "borderTopWidth": 4,
        "display": "flex",
        "height": 252.5,
        "justifyContent": "center",
        "paddingBottom": 40,
        "paddingLeft": 24,
        "paddingRight": 24,
        "paddingTop": 28,
        "shadowColor": "#000",
        "shadowOffset": Object {
          "height": 2,
          "width": 0,
        },
        "shadowOpacity": 0.05,
        "shadowRadius": 3,
      },
    ]
  }
  theme={
    Object {
      "backgroundColor": Object {
        "default": "#f9f9f9",
      },
      "borders": Array [
        0,
        "1px solid",
        "2px solid",
      ],
      "breakpoints": Array [
        "420px",
        "740px",
        "980px",
        "1300px",
      ],
      "checkBox": Object {
        "checked": "#0275d8",
        "text": "#212121",
        "unchecked": "#9a9a9a",
      },
      "colors": Object {
        "amber": Array [
          "#FF9900",
        ],
        "backgroundColor": "#f9f9f9",
        "black": "#000000",
        "blue": Array [
          "#0275d8",
          "#afdaff",
          "#3a98c9",
        ],
        "buttonPrimary": "#50ABAD",
        "buttonPrimaryPressed": "#AD0012",
        "error": Array [
          "#b00020",
          "#f6710a",
          "#fcd1b9",
          "#d7d8d6",
        ],
        "fonts": Object {
          "link": "#0275D8",
        },
        "gray": Array [
          "#212121",
          "#646464",
          "#727272",
          "#9a9a9a",
          "#b8b8b8",
          "#e3e3e3",
          "#f0f0f0",
          "#f9f9f9",
          "#666666",
          "#f2f2f2",
          "#ededed",
        ],
        "green": Array [
          "#00847F",
        ],
        "imageHeaderMatching": "#f2f2f2",
        "info": Array [
          "#31707e",
          "#1d5b6a",
          "#d9edf7",
        ],
        "primary": Array [
          "#50ABAD",
          "#AD0012",
        ],
        "red": Array [
          "#DB0011",
        ],
        "riskAmber": "#FF9900",
        "riskGreen": "#00847F",
        "riskRed": "#DB0011",
        "secondary": Array [
          "#2f3494",
          "#648be7",
          "#cfe1ff",
        ],
        "success": Array [
          "#03A450",
          "#77b03c",
          "#dff0d8",
        ],
        "touchableOverlayColor": "#f0f0f0",
        "transparent": "transparent",
        "warning": Array [
          "#935c01",
          "#ffae28",
          "#fdf1b2",
        ],
        "white": "#ffffff",
      },
      "fontSizes": Array [
        12,
        14,
        16,
        20,
        24,
        32,
        48,
        64,
        96,
        128,
      ],
      "fontWeights": Object {
        "bold": 700,
        "bolder": 800,
        "light": 300,
        "normal": 500,
      },
      "fonts": Object {
        "default": "Open Sans",
      },
      "healthScoreGraphicColors": Object {
        "averageScore": "#ffae28",
        "default": "#e3e3e3",
        "highScore": "#03A450",
        "lowScore": "#50ABAD",
        "none": "transparent",
      },
      "inputField": Object {
        "disabledBorder": "#d1d1d1",
        "disabledText": "rgba(0, 0, 0, 0.38)",
        "inputBorder": "#000000",
        "inputBorderUntouched": "#9a9a9a",
        "inputFocus": "#0275d8",
        "inputText": "#646464",
        "rightText": "rgba(0, 0, 0, 0.38)",
      },
      "letterSpacings": Object {
        "caps": "0.25em",
        "normal": "normal",
      },
      "lineHeights": Array [
        1,
        1.125,
        1.25,
        1.5,
        20,
        37,
      ],
      "radii": Array [
        0,
        2,
        4,
        8,
      ],
      "radioButton": Object {
        "checked": "#00847f",
      },
      "shadows": Array [
        "0 1px 2px 0 #9a9a9a",
        "0 1px 4px 0 #9a9a9a",
      ],
      "space": Array [
        0,
        4,
        8,
        16,
        32,
        64,
        128,
      ],
      "tabbar": Object {
        "activeColor": "#DB0011",
        "activeTextColor": "#DB0011",
        "inactiveColor": "#bdbdbd",
      "terminatedLabel": Object {
        "backgroundColor": "rgba(233, 161, 21, 0.1)",
      },
    }
  }
>
  <Text
    accessibilityLabel="Unable to show data."
    accessible={true}
    color="gray.8"
    flexGrow={1}
    fontFamily="default"
    fontSize={14}
    letterSpacing={0.3}
    lineHeight={22}
    style={
      Array [
        Object {
          "color": "#666666",
          "fontFamily": "Open Sans",
          "fontSize": 14,
          "letterSpacing": 0.3,
          "lineHeight": 22,
        },
      ]
    }
  >
    Unable to show data.
  </Text>
  <Text
    accessibilityLabel="Try again later."
    accessible={true}
    color="gray.8"
    flexGrow={1}
    fontFamily="default"
    fontSize={14}
    letterSpacing={0.3}
    lineHeight={22}
    style={
      Array [
        Object {
          "color": "#666666",
          "fontFamily": "Open Sans",
          "fontSize": 14,
          "letterSpacing": 0.3,
          "lineHeight": 22,
        },
      ]
    }
  >
    Try again later.
  </Text>
</View>
`;
